resources/lib/kodiutils.py

```diff
@@ -697,24 +697,19 @@ def get_cache(path, ttl=None):  # pylint: disable=redefined-outer-name
     mtime = stat_file(fullpath).st_mtime()
     now = mktime(localtime())
     if ttl is None or now - mtime < ttl:
-        try:  # Python 3
-            from json import loads, JSONDecodeError
-            ValueError = BaseException  # pylint: disable=invalid-name, redefined-builtin
-        except ImportError:  # Python 2
-            from json import loads
-            JSONDecodeError = BaseException
+        from json import loads
         if ttl is None:
             log(3, "Cache '{path}' is forced from cache.", path=path)
         else:
             log(3, "Cache '{path}' is fresh, expires in {time}.", path=path, time=human_delta(mtime + ttl - now))
         cache_data = None
         with open_file(fullpath, 'r') as fdesc:
-            try:
-                cache_data = to_unicode(fdesc.read())
-                if cache_data:
+            cache_data = to_unicode(fdesc.read())
```

Comment:
I don't know if you intended to also catch exceptions from to_unicode() or read() as well.
If so, we should use a different try-except block.

It's always best to use different try-blocks for each statement.

Reply:
No, exceptions only occur when passing the cache_data to the json module.
```
2019-12-08 22:32:19.050 T:140696274335488   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.ValueError'>
                                            Error Contents: Expecting object: line 1 column 94528 (char 94527)
                                            Traceback (most recent call last):
                                              File "/home/test/.kodi/addons/plugin.video.vrt.nu/resources/lib/addon_entry.py", line 14, in <module>
                                                run(argv)
                                              File "/home/test/.kodi/addons/plugin.video.vrt.nu/resources/lib/addon.py", line 298, in run
                                                plugin.run(argv)
                                              File "/home/test/.kodi/addons/script.module.routing/lib/routing.py", line 130, in run
                                                self._dispatch(self.path)
                                              File "/home/test/.kodi/addons/script.module.routing/lib/routing.py", line 141, in _dispatch
                                                view_func(**kwargs)
                                              File "/home/test/.kodi/addons/plugin.video.vrt.nu/resources/lib/addon.py", line 140, in resumepoints_watchlater
                                                VRTPlayer().show_watchlater_menu(page=1)
                                              File "/home/test/.kodi/addons/plugin.video.vrt.nu/resources/lib/vrtplayer.py", line 303, in show_watchlater_menu
                                                self._resumepoints.refresh(ttl=ttl('direct'))
                                              File "/home/test/.kodi/addons/plugin.video.vrt.nu/resources/lib/resumepoints.py", line 37, in refresh
                                                resumepoints_json = get_cache('resume_points.json', ttl)
                                              File "/home/test/.kodi/addons/plugin.video.vrt.nu/resources/lib/kodiutils.py", line 707, in get_cache
                                                return load(fdesc)
                                              File "/usr/lib/python2.7/json/__init__.py", line 291, in load
                                                **kw)
                                              File "/usr/lib/python2.7/json/__init__.py", line 339, in loads
                                                return _default_decoder.decode(s)
                                              File "/usr/lib/python2.7/json/decoder.py", line 364, in decode
                                                obj, end = self.raw_decode(s, idx=_w(s, 0).end())
                                              File "/usr/lib/python2.7/json/decoder.py", line 380, in raw_decode
                                                obj, end = self.scan_once(s, idx)
                                            ValueError: Expecting object: line 1 column 94528 (char 94527)

```

This exception happens after playing a video for more than 5 minutes and stopping the video.

Reply:
Hmmm, that is weird.
We ought to dump that content :-/

Reply:
Maybe we are having issues with `update_cache()`, `json.dump()` and encoding?

Reply:
No, I tested and `resume_points.json` on disk is always a valid json file.